lib/assignDisabledRanges.mjs

```diff
@@ -268,34 +345,58 @@ export default function assignDisabledRanges(root, result) {
 				continue;
 			}
 
-			throw comment.error(`"${ruleToEnable}" has not been disabled`, {
+			throw node.error(`"${ruleToEnable}" has not been disabled`, {
 				plugin: 'stylelint',
 			});
 		}
 	}
 
 	/**
-	 * @param {PostcssComment} comment
+	 * @param {PostcssNode} node
+	 * @param {string} part
+	 * @param {number} offset
+	 */
+	function checkCommentContainingNode(node, part, offset) {
+		tokenize({ css: part })
+			.filter((token) => {
+				return isTokenComment(token);
+			})
+			.forEach((token) => {
+				const [, text, start, end] = token;
+
+				const source = node.rangeBy({
+					index: start + offset,
+					endIndex: end + 1 + offset,
+				});
+
+				checkComment(node, source, text.slice(2, -2).trim());
+			});
+	}
+
+	/**
+	 * @param {PostcssNode} node
+	 * @param {Source | undefined} source
+	 * @param {string} text
 	 */
-	function checkComment(comment) {
+	function checkComment(node, source, text) {
```

Comment:
[ask] Is the following early return for `source` effective? It seems that we can skip `processDisable*()` function calls if `source` is undefined. 🤔 

```suggestion
	function checkComment(node, source, text) {
		if (!source) return;
```

Reply:
yeah, that makes sense :)